collection-scripts/gather_ppc

```diff
@@ -0,0 +1,203 @@
+#!/bin/bash
+
+BASE_COLLECTION_PATH="must-gather"
+NODES_PATH=${BASE_COLLECTION_PATH}/nodes
+
+# Once you start the pod, the Kubernetes will set the pod hostname to the name of the pod
+# https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields
+NAMESPACE=$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)
+
+POD_NAME=${HOSTNAME}
+POD_IP=$(hostname -I |  tr -d "[:blank:]" )
+
+function check_node_gather_pods_ready() {
+  line=$(oc get ds perf-node-gather-daemonset -o=custom-columns=DESIRED:.status.desiredNumberScheduled,READY:.status.numberReady --no-headers -n $NAMESPACE)
+
+  IFS=$' '
+  read desired ready <<< $line
+  IFS=$'\n'
+
+  if [[ "$desired" != "0" ]] && [[ "$ready" == "$desired" ]]
+  then
+    echo "Daemonset perf-node-gather-daemonset ready $ready out of $desired"
+    return 0
+  else
+    return 1
+  fi
+}
+
+# Start a specially privileged pod on each node and collect node level performance tuning data
+function ppc_nodes() {
+  IFS=$'\n'
+
+  # Create the destination
+  mkdir -p ${NODES_PATH}
+
+  # Save the debug pod info
+  echo "[$NAMESPACE/$POD_IP/$POD_NAME]" >> ${NODES_PATH}/debug
+  oc get pod -n $NAMESPACE $POD_NAME -o json >> ${NODES_PATH}/debug
+
+  # Find the NTO image reference
+  # NTO contains all the tools needed here
+  NTO=$(oc adm release info --image-for=cluster-node-tuning-operator)
```

Comment:
I'd add error checking here, since it's pointless to proceed if this commands fails.